plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -413,6 +419,23 @@ public RemoteIterator<LocatedFileStatus> listLocatedStatus(Path path)
         return new S3ObjectsV2RemoteIterator(listPath(path, OptionalInt.empty(), ListingMode.SHALLOW_ALL));
     }
 
+    @Override
+    public void batchDeleteFiles(Collection<Path> paths)
+            throws IOException
+    {
+        try {
+            List<KeyVersion> keys = paths.stream().map(TrinoS3FileSystem::keyFromPath).map(KeyVersion::new).collect(toImmutableList());
+            DeleteObjectsRequest deleteObjectsRequest = new DeleteObjectsRequest(getBucketName(uri))
```

Comment:
There is a max number of keys allowed per request, if you have more than that you need to split them. There should be a constant defined somewhere but it's 1000 keys https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObjects.html

Reply:
Thanks. I am aware of this.  For now I did this filtering in IcebergMetadata, I will move it here,